### PR TITLE
VACMS-10266 Add preliminary next build integration

### PR DESCRIPTION
**Pull request by timcosgrove**
## Description
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/10266

When we launch next-build, the initial phase will make use of content-build to fire off next-build running in parallel, and then pull the generated files back into content-build for further processing and deployment.

This PR adds the plugins and commands necessary to run this. They will be wrapped in checks for options and only run with those options active, and never on prod builds (for the time being).

## Testing done
- [x] next-build is no-op when running on Tugboat
- [x] next-build can be added to content-build locally given the proper flags & env variables

Future tickets will set up next-build to run on Tugboat given the correct input. Other environments (Local dev, CI & build) will be enabled once next-build is considered ready for use.

### CODE REVIEW

File: script/github-actions/check-broken-links-blocks.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ const reportPath = `./logs/${envName}-broken-links.json`;
 const SERVER_URL = `${process.env.GITHUB_SERVER_URL}/${process.env.GITHUB_REPOSITORY}/actions/runs/${process.env.GITHUB_RUN_ID}`;
 const BRANCH_NAME = process.env.GITHUB_REF;
 const IS_PROD_BRANCH = BRANCH_NAME.replace('refs/heads/', '') === 'main';
-const GITHUB_WORKFLOW = process.env.GITHUB_WORKFLOW;
+const { GITHUB_WORKFLOW } = process.env;
```
Comment (timcosgrove) [inline script/github-actions/check-broken-links-blocks.js:11]: eslint was complaining about this.

File: script/github-actions/check-broken-links.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ if (fs.existsSync(reportPath)) {
     brokenLinks.brokenLinksCount >
       (brokenLinks.maxBrokenLinks ?? maxBrokenLinks);
   const color = shouldFail ? '#D33834' : '#FFCC00'; // danger or warning, needs to be in hex
-  const summary = brokenLinks.summary;
+  const { summary } = brokenLinks;
```
Comment (timcosgrove) [inline script/github-actions/check-broken-links.js:23]: eslint was complaining about this.

File: src/site/stages/build/index.js [hunk 3]
```diff
@@ -231,6 +233,11 @@ function build(BUILD_OPTIONS) {
     );
   });
 
+  smith.use(
+    addDirectoryFiles(`${BUILD_OPTIONS.nextBuildDirectory}out/`, true),
```
Comment (timcosgrove) [inline src/site/stages/build/index.js:237]: todo: parameterize the output directory.

File: src/site/tests/support/index.js [hunk 4]
```diff
@@ -4,14 +4,14 @@ import path from 'path';
 import cheerio from 'cheerio';
 import { JSDOM } from 'jsdom';
 import liquid from 'tinyliquid';
+import ENVIRONMENT_CONFIGURATIONS from 'site/constants/environments-configs';
```
Comment (timcosgrove) [inline src/site/tests/support/index.js:7]: eslint doing its thing.

File: src/site/stages/build/drupal/graphql/GetAllPages.graphql.js [hunk 5]
```diff
@@ -49,7 +49,7 @@ const { cmsFeatureFlags } = global;
 const {
   updateQueryString,
   queryParamToBeChanged,
-} = require('./../../../../utilities/stringHelpers');
```
Comment (timcosgrove) [inline src/site/stages/build/drupal/graphql/GetAllPages.graphql.js:52]: eslint doing its thing.

File: script/install-repos.sh [hunk 6]
```diff
@@ -16,3 +16,12 @@ if [ ! -d ../vets-api ]; then
 else
   echo "Repo vets-api already cloned."
 fi
+
+if [ "${RUN_NEXT_BUILD}" = "1" ]; then
```
Comment (timcosgrove) [inline script/install-repos.sh:20]: Ensuring this only happens when we really want it to.

File: src/site/stages/build/index.js [hunk 7]
```diff
@@ -87,8 +89,11 @@ function build(BUILD_OPTIONS) {
     );
   }
 
-  smith.use(generateStaticDataFiles(BUILD_OPTIONS), 'Build static data files');
+  if (BUILD_OPTIONS.runNextBuild && BUILD_OPTIONS.buildtype !== 'vagovprod') {
```
Comment (timcosgrove) [inline src/site/stages/build/index.js:92]: needs `--run-next-build` and to not be a prod build. Never do this on prod until we're 100% ready.

File: src/site/stages/build/plugins/add-directory-files.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
+/* eslint-disable no-param-reassign */
```
[the remark this answers is not in the document]
Comment (timcosgrove) [reply src/site/stages/build/plugins/add-directory-files.js:1]: Metalsmith plugins require manipulation of the `files` argument that is passed to them.

File: src/site/stages/build/plugins/add-directory-files.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+/* eslint-disable no-console */
```
[the remark this answers is not in the document]
Comment (timcosgrove) [reply src/site/stages/build/plugins/add-directory-files.js:2]: We're logging. Sorry!

File: src/site/stages/build/plugins/run-next-build.js [hunk 10]
```diff
@@ -0,0 +1,63 @@
+/* eslint-disable no-param-reassign */
```
[the remark this answers is not in the document]
Comment (timcosgrove) [reply src/site/stages/build/plugins/run-next-build.js:1]: Same as above.

File: src/site/stages/build/plugins/run-next-build.js [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+/* eslint-disable no-console */
```
[the remark this answers is not in the document]
Comment (timcosgrove) [reply src/site/stages/build/plugins/run-next-build.js:2]: Same as above.

**Review by tjheffner — APPROVED** [review]

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Pull request merged**